### ISSUE TITLE
Fix integration test "FreeText Editor FreeText (delete existing) must delete an existing annotation"

### ISSUE THREAD

**Issue by timvandermeij** [mozilla/pdf.js#18368]
The most recent failure is in http://54.193.163.58:8877/018387cde149e38/output.txt with the following traceback:

```
Failures:
1) FreeText Editor FreeText (delete existing) must delete an existing annotation
  Message:
[31m    ProtocolError: Waiting for selector `#pdfjs_internal_editor_3.selectedEditor` failed: script.callFunction timed out. Increase the 'protocolTimeout' setting in launch/connect calls for a higher timeout if needed.[0m
  Stack:
        at <instance_members_initializer> (file:///C:/pdfjs/botio-files-pdfjs/private/018387cde149e38/node_modules/puppeteer-core/lib/esm/puppeteer/common/CallbackRegistry.js:89:14)
        at new Callback (file:///C:/pdfjs/botio-files-pdfjs/private/018387cde149e38/node_modules/puppeteer-core/lib/esm/puppeteer/common/CallbackRegistry.js:93:16)
        at CallbackRegistry.create (file:///C:/pdfjs/botio-files-pdfjs/private/018387cde149e38/node_modules/puppeteer-core/lib/esm/puppeteer/common/CallbackRegistry.js:19:26)
        at BidiConnection.send (file:///C:/pdfjs/botio-files-pdfjs/private/018387cde149e38/node_modules/puppeteer-core/lib/esm/puppeteer/bidi/Connection.js:51:32)
        at Session.send (file:///C:/pdfjs/botio-files-pdfjs/private/018387cde149e38/node_modules/puppeteer-core/lib/esm/puppeteer/bidi/core/Session.js:154:42)
        at Session.<anonymous> (file:///C:/pdfjs/botio-files-pdfjs/private/018387cde149e38/node_modules/puppeteer-core/lib/esm/puppeteer/util/decorators.js:94:27)
        at WindowRealm.callFunction (file:///C:/pdfjs/botio-files-pdfjs/private/018387cde149e38/node_modules/puppeteer-core/lib/esm/puppeteer/bidi/core/Realm.js:92:51)
        at WindowRealm.<anonymous> (file:///C:/pdfjs/botio-files-pdfjs/private/018387cde149e38/node_modules/puppeteer-core/lib/esm/puppeteer/util/decorators.js:94:27)
        at #evaluate (file:///C:/pdfjs/botio-files-pdfjs/private/018387cde149e38/node_modules/puppeteer-core/lib/esm/puppeteer/bidi/Realm.js:127:42)
        at async BidiFrameRealm.evaluateHandle (file:///C:/pdfjs/botio-files-pdfjs/private/018387cde149e38/node_modules/puppeteer-core/lib/esm/puppeteer/bidi/Realm.js:94:16)
```

The source code can be found here: https://github.com/mozilla/pdf.js/blob/master/test/integration/freetext_editor_spec.mjs#L1213-L1263

We should find out what causes this as it's one of the last intermittents we have left.

**Comment by calixteman** [comment]
Maybe I fixed it in https://github.com/mozilla/pdf.js/pull/18366.
The rect is empty because the editor isn't rendered and then clicking on "it" does nothing, consequently the editor isn't selected.

**Comment by timvandermeij** [comment]
Unfortunately I don't think that fixed this particular issue because this intermittent happened in #18367 and that one was rebased onto the current master branch where that patch is already active.

**Comment by timvandermeij** [comment]
It's indeed a bit problematic because I can't spot anything wrong in the code, especially after the extra hardening that you applied to `getRect`, and I can't reproduce it locally too. For now let's just use this issue to track that it happened and see if it happens again, and in the meantime I'll think about (and have some ideas for) how to get more debugging information if such exceptions happen.

**Comment by timvandermeij** [comment]
For later reference, it just happened again in #18369, see http://54.193.163.58:8877/fc78ad56a1a8b11/output.txt, which also contains the `getRect` hardening, so it does appear that there is a different issue with this particular test only.

**Comment by timvandermeij** [comment]
This one might be fixed by #18374 but because it's a bit hard to tell due to its intermittent nature we'll leave this tracking issue open for a couple of days to make sure it doesn't reappear.

**Comment by timvandermeij** [comment]
Closing since we haven't seen this reappear so far. If it does, we'll track it in #18396.
